tests/plugins/link/anchor.js

```diff
@@ -79,6 +79,46 @@
 					assert.areSame( anchor, dialog.getModel( editor ) );
 				} );
 			} );
+		},
+
+		// (#3863)
+		'test preserve duplicate anchors after editing word with custom style': function() {
```

Comment:
```suggestion
		'test prevent duplicated anchors after editing a word with custom style': function() {
```

?

Reply:
Could you add another similar test with more nesting like e.g. `strong > i > span`?